rts/motoko-rts/src/blob_iter.rs

```diff
@@ -38,14 +39,14 @@ unsafe extern "C" fn blob_iter_done(iter: Value) -> u32 {
 }
 
 /// Reads next byte, advances the iterator
-#[no_mangle]
-unsafe fn blob_iter_next(iter: Value) -> u32 {
+#[ic_mem_fn]
+unsafe fn blob_iter_next<M: Memory>(mem: &mut M, iter: Value) -> u32 {
     let iter_array = iter.as_array();
 
     let blob = iter_array.get(ITER_BLOB_IDX);
     let pos = iter_array.get(ITER_POS_IDX).get_scalar();
 
-    iter_array.set_scalar(ITER_POS_IDX, Value::from_scalar(pos + 1));
+    iter_array.set(ITER_POS_IDX, Value::from_scalar(pos + 1), mem);
```

Comment:
(another untagged scalar)

Reply:
See above. This seems safe: https://github.com/dfinity/motoko/pull/4562